packages/@sanity/default-layout/src/navbar/search/SearchField.tsx

```diff
@@ -1,196 +1,67 @@
-                    {error?.message || 'Something went wrong while searching'}
-                  </StyledText>
-                </Flex>
-              </Box>
-            }
-          />
-        )
-      }
-
-      if (!popoverProps.hidden && searchString && !loading && hits.length === 0) {
-        return (
-          <SearchPopoverContent
-            open={!popoverProps.hidden}
-            referenceElement={popoverProps.inputElement}
-            ref={ref}
-            content={
-              <Box padding={4}>
-                <Flex align="center" height="fill" justify="center">
-                  <StyledText align="center" muted>
-                    No results for <strong>“{searchString}”</strong>
-                  </StyledText>
-                </Flex>
-              </Box>
-            }
-          />
-        )
-      }
-
-      if (!popoverProps.hidden && hits.length > 0) {
-        return (
-          <SearchPopoverContent
-            open={!popoverProps.hidden}
-            referenceElement={popoverProps.inputElement}
-            ref={ref}
-            content={popoverProps.content}
-          />
-        )
-      }
-
-      return undefined
-    },
-    [error, searchString, loading, hits]
-  )
-
-  const autoComplete = useMemo(
-    () => (
-      <Autocomplete
-        filterOption={filterOption}
-        icon={SearchIcon}
-        id="studio-search"
-        key="studio-search"
-        listBox={{padding: fullScreen ? 2 : 1}}
-        loading={loading}
-        onQueryChange={handleSearch}
-        options={hits.map((hit) => {
-          return {
-            value: hit.hit._id,
-            payload: {
-              data: hit,
-            },
-          }
-        })}
-        placeholder="Search"
-        radius={2}
-        ref={inputElement}
-        relatedElements={relatedElements}
-        renderOption={renderOption}
-        renderPopover={fullScreen ? renderPopoverFullscreen : renderPopover}
+import React, {useCallback, useEffect, useState} from 'react'
+import {PlaceholderSearchInput} from './components/PlaceholderSearchInput'
+import {SearchPopover} from './components/SearchPopover'
+import {POPOVER_INPUT_PADDING, POPOVER_MAX_WIDTH} from './constants'
+
+export function SearchField() {
+  const [placeholderElement, setPlaceholderElement] = useState<HTMLInputElement | null>(null)
+  const [open, setOpen] = useState(false)
+  const [popoverPosition, setPopoverPosition] = useState<{
+    x: number
+    y: number
+  }>(null)
+
+  const handleClose = useCallback(() => setOpen(false), [])
+  const handleOpen = useCallback(() => setOpen(true), [])
+
+  /**
+   * Update popover position on window resize based off current placeholder input
+   */
+  const handleWindowResize = useCallback(() => {
+    setPopoverPosition(calcDialogPosition(placeholderElement))
+  }, [placeholderElement])
+
+  useEffect(() => {
+    setPopoverPosition(calcDialogPosition(placeholderElement))
+  }, [placeholderElement])
+
+  useEffect(() => {
```

Comment:
Suggestion: create a usePopoverPosition hook, move the sideefects and state into it and return popoverPosition; this will make the callsite here a bit more tidy?

Reply:
Good shout, that's in!